extras/CatchAddTests.cmake

```diff
@@ -121,9 +115,18 @@ function(catch_discover_tests_impl)
     endforeach()
   endif()
 
+  # Catch2 string escape logic is a bit funky, double quotes are escaped but single
+  # slash is not, the following line of code tries to handle it but it can be brittle.
+  string(REGEX REPLACE [[\\([^"])]] [[\\\\\1]] output "${output}")
```

Comment:
I don't understand what this is trying to do. Can you provide examples?

Also the escape happening before the JSON is parsed is suspect.

Reply:
This test fails without manually escaping the backlash
```cpp
TEST_CASE("@Script[C:\\EPM1A]=x;\"SCALA_ZERO:\"", "[script regressions]"){}
```
basically, double quotes are escaped, but `C:\E` is output as is hence is interpreted as `\E`

Reply:
>Also the escape happening before the JSON is parsed is suspect.
Yes, that's why I comment the escape logic is funky, the escape logic in Catch itself should be revisited

Reply:
That's an escaping bug in the JsonWriter.

Currently it outputs this
```
        "name": "@Script[C:\EPM1A]=x;\"SCALA_ZERO:\"",
```

but it should output this 
```
        "name": "@Script[C:\\EPM1A]=x;\"SCALA_ZERO:\"",
```

I already fixed it in a branch, I can merge it into main tomorrow.

Reply:
oh, it was my bad then, sorry for the comment blaming Catch2.